Framework/DataHandling/src/MaskDetectors.cpp

```diff
@@ -172,17 +172,35 @@ void MaskDetectors::exec() {
         appendToIndexListFromMaskWS(indexList, maskWS, ranges_info);
       }
     } else { // not mask workspace
+      auto nHist = prevMasking->getNumberHistograms();
       // Check the provided workspace has the same number of spectra as the
       // input
-      if (prevMasking->getNumberHistograms() > WS->getNumberHistograms()) {
-        g_log.error() << "Input workspace has " << WS->getNumberHistograms()
-                      << " histograms   vs. "
-                      << "Input masking workspace has "
-                      << prevMasking->getNumberHistograms()
-                      << " histograms. \n";
-        throw std::runtime_error("Size mismatch between two input workspaces.");
+      if (nHist > WS->getNumberHistograms()) {
+        // Check the provided workspace has the same number of detectors as the
+        // input. If so add to the detector list
+        auto instrument = WS->getInstrument();
+        auto maskInstrument = prevMasking->getInstrument();
+        if (instrument && maskInstrument &&
+            maskInstrument->getNumberDetectors() ==
+                instrument->getNumberDetectors() &&
```

Comment:
if the number of detectors in the instruments do not match we should just warn but continue, not refuse.